os/kernel/log_dump/log_dump.c

```diff
@@ -141,13 +141,13 @@ int log_dump_init(void)
 	 * device is shut down or restarted. So these buffers MUST NEVER
 	 * BE FREED. 
 	 */
-	out_buf = allocate_compress_buffer(0, CONFIG_LOG_DUMP_CHUNK_SIZE);
+	out_buf = (unsigned char *)allocate_compress_buffer(0, CONFIG_LOG_DUMP_CHUNK_SIZE);
```

Comment:
why do we need to typecast here, if allocate_compress_buffer is already returning the pointer to unsigned char ?

Reply:
it is found that allocate_compress_buffer was not declared in the header file and will be resolved in the next commit